### PR TITLE
man: document signal unsafeness of sd_journal_get_fd

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: man/sd_journal_get_fd.xml [hunk 1]
```diff
@@ -230,6 +230,20 @@ else {
     journal.</para>
   </refsect1>
 
+  <refsect1>
+    <title>Signal safety</title>
+
+    <para>In general, <function>sd_journal_get_fd()</function>,
+    <function>sd_journal_get_fd()</function>, and <function>sd_journal_get_timeout()</function> are
```
Comment (yuwata) [inline man/sd_journal_get_fd.xml]: `sd_journal_get_fd()` is duplicated.

File: man/sd_journal_get_fd.xml [hunk 2]
```diff
@@ -230,6 +230,20 @@ else {
...
+    <emphasis>not</emphasis> "async signal safe" in the meaning of <citerefentry
+    project='man-pages'><refentrytitle>signal</refentrytitle><manvolnum>7</manvolnum></citerefentry>.
+    Nevertheless, only the first call to any of those three functions performs unsafe operations, so
+    subsequent calls <emphasis>are</emphasis> safe.</para>
+
+    <para><function>sd_journal_process</function> and <function>sd_journal_wait</function> are not
+    safe. <function>sd_journal_reliable_fd</function> is safe.</para>
```
Comment (yuwata) [inline man/sd_journal_get_fd.xml]: missing `()`.

**Comment by keszybz** [comment]
Thanks, updated.

**Comment by yuwata** [comment]
lgtm.

**Comment by sourcejedi** [comment]
This closes #7912, which is specifically about sd_journal_stream_fd().  But this doesn't mention sd_journal_stream_fd().  What am I missing?

**Comment by sourcejedi** [comment]
> Oops. There's nothing wrong with #7938, but it documents the wrong function ;)

Sorry, I wrote this as a comment edit, so it might not have reached you.

> There's a specific reason sd_journal_stream_fd() would be useful as a signal-safe function. I don't think it applies to these cases; there doesn't seem a reason to make a commitment to async signal safety (or mention the lack of it).

That is, I think it would also be best to revert this PR (#7938).

**Comment by keszybz** [comment]
> Sorry, I wrote this as a comment edit, so it might not have reached you.

This PR was in response to your comment.

> That is, I think it would also be best to revert this PR

I don't think it hurts in any way so I don't see a particular reason to revert it.

**Comment by sourcejedi** [comment]
It's extra text for the reader.  It implies that it this information is relevant to something.  Reading it, I try to work out what the reason is for it being relevant.  I.e. why I want to know that some very specific uses of these functions are async-signal safe, but some are not.  But there is no reason for it to be relevant.  Therefore it is confusing to the reader.

**Pull request merged**